cmd/cluster/dynatrace/requests.go

```diff
@@ -186,35 +205,93 @@ func getRequestToken(query string, dtURL string, accessToken string) (requestTok
 	return execResp.RequestToken, nil
 }
 
-func getLogs(dtURL string, accessToken string, requestToken string) error {
-	reqData := url.Values{
-		"request-token": {requestToken},
-	}.Encode()
-
-	requester := Requester{
-		method: http.MethodGet,
-		url:    dtURL + "platform/storage/query/v1/query:poll?" + reqData,
-		headers: map[string]string{
-			"Content-Type":  "application/json",
-			"Authorization": "Bearer " + accessToken,
-		},
-		successCode: http.StatusOK,
+func getDTPollResults(dtURL string, requestToken string, accessToken string) (respBody string, error error) {
+	var requester Requester
+	var dtPollRes DTLogsPollResult
+
+	for {
+		reqData := url.Values{
+			"request-token": {requestToken},
+		}.Encode()
+
+		// avoid creating multiple instances
+		if requester.url == "" {
+			requester = Requester{
+				method: http.MethodGet,
+				url:    dtURL + "platform/storage/query/v1/query:poll?" + reqData,
+				headers: map[string]string{
+					"Content-Type":  "application/json",
+					"Authorization": "Bearer " + accessToken,
+				},
+				successCode: http.StatusOK,
+			}
+		}
+		resp, err := requester.send()
+		if err != nil {
+			return "", err
+		}
+
+		err = json.Unmarshal([]byte(resp), &dtPollRes)
+		if err != nil {
+			return "", err
+		}
+
+		if dtPollRes.State == "RUNNING" {
+			continue
+		}
+
+		if dtPollRes.State == "SUCCEEDED" {
+			return resp, nil
+		}
+
+		if dtPollRes.State != "RUNNING" && dtPollRes.State == "SUCCEEDED" {
```

Comment:
This if-statement can be simplified to just the second clause
```suggestion
		if dtPollRes.State == "SUCCEEDED" {
```